### DISCUSSION TITLE
Is it possible to retrieve the current validation mode?

### DISCUSSION THREAD

**Discussion by mbrowne** [react-hook-form/react-hook-form#11824] [Q&A]
I have a custom Select component that requires triggering validation manually (it does have a ref, but the ref is just attached to a hidden input to track the value). I would like to add event handlers like this to the wrapper for my component, but how can I get the `mode` variable?

```
        onBlur={() => {
            if (mode === 'onBlur') {
                trigger(field.name)
            }
        }}
```

I was hoping I could find the current mode on the object returned from `useFormContext()`, but it doesn't seem to be there.

**Comment by larrybarriosjr** [comment]
I needed this functionality, too, so I tried looking for it. You might have found it already, but for those who are looking and ended up here. Here's where I found it:

```ts
const { control } = useFormContext()
control._options.mode // "all" | "onBlur" | "onChange" | "onSubmit" | "onTouched"
```
